In [ ]:
from scipy.io import loadmat

# Exploring File Structure 
data = loadmat('../data/train.mat')

print([k for k in data.keys() if not k.startswith('__')])

['trainData', 'trainLabels']


In [2]:
train_data = data['trainData']
train_labels = data['trainLabels']

print(f"trainData: {train_data.shape, train_data.dtype}")
print(f"trainLabels: {train_labels.shape, train_labels.dtype}")

trainData: ((393, 5000), dtype('<f8'))
trainLabels: ((393, 1), dtype('O'))


In [ ]:
# observe first 5 labels to clarify type
print(f"First 5 labels: {train_labels[:5]}")

# find type of just one label object
print(f"Type of first label: {type(train_labels[0,0])}")

First 5 labels: [[array(['InnerRaceFault'], dtype='<U14')]
 [array(['InnerRaceFault'], dtype='<U14')]
 [array(['InnerRaceFault'], dtype='<U14')]
 [array(['InnerRaceFault'], dtype='<U14')]
 [array(['InnerRaceFault'], dtype='<U14')]]
Type of first label: <class 'numpy.ndarray'>


The train.mat dataset has 393 examples. 

There are 5000 float64 features and 1 label of this structure: []

In [ ]:
# capture unwrapped element
unwrapped_label = train_labels[0,0][0]
print(f"Label: {unwrapped_label}")

# capture type of unwrapped element type
print(f"Label Type: {type(unwrapped_label[0])}")

Label: InnerRaceFault
Label Type: <class 'str'>


The Label is a String representing what (if any) kind of fault there is

In [ ]:
import numpy as np

